File: app/src/main/java/com/newchar/woolhelper/db/SQLHelper.java
package com.newchar.woolhelper.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

import com.newchar.woolhelper.entry.HasOrderAppItem;

/**
 * @author newChar
 * date 2021/6/18
 * @since 当前版本，（以及描述）
 * @since 迭代版本，（以及描述）
 */
public class SQLHelper extends SQLiteOpenHelper {

    /**
     * 初始化版本
     */
    private static final int VERSION_INIT = 1;

    public SQLHelper(Context context) {
        super(context, "cmd", null, VERSION_INIT);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        try {
            db.beginTransaction();
            db.execSQL(HasOrderAppItem.TABLE.createTableSql());
            db.setTransactionSuccessful();
            Log.d("SQLHelper", " cmd 开始创建");
        } finally {
            db.endTransaction();
            Log.d("SQLHelper", "cmd 创建完成");
        }
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }

}

File: app/src/main/java/com/newchar/woolhelper/db/SQLHelper.kt
package com.newchar.woolhelper.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

/**
 *  @author wenliqiang
 *  date 2019-12-01
 *  @since 当前版本，（以及描述）
 *  @since 迭代版本，（以及描述）
 */
class SQLHelper(var ctx: Context, var version: Int) : SQLiteOpenHelper(ctx, DB_NAME, null, version) {

    val VERSION_DEFAULT = 1
    val VERSION_DEFAULdT = 2

    companion object{
        val NAME = "ACTION_NAME"
        val DB_NAME = "AAAA"
        val CREATE_SQL_TABLE = "CREATE TABLE ${NAME} " +"{" + "_id varchar(30), id integer(10)" + "}"
    }

    override fun onCreate(db: SQLiteDatabase?) {
        try {
            db?.run {
                beginTransaction()
                execSQL(CREATE_SQL_TABLE)
            }
        } finally {
            db?.run {
                setTransactionSuccessful()
                close()
            }
        }
    }

    override fun onDowngrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {}

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
        when (newVersion) {
            VERSION_DEFAULT->{

            }
            VERSION_DEFAULdT -> {
            }
        }
    }

}